errors/FoundError.js

```diff
@@ -0,0 +1,10 @@
+function FoundError(error, res) {
+  if (error.name === 'ValidationError') {
+    return res.status(400).send({ message: error.message });
+  }
+  if (error.name === 'CastError') {
+    return res.status(400).send({ message: 'id is not found' });
+  }
+  return res.status(500).send({ message: error.message });
```

Comment:
Нужно исправить: при бросании исключения с экземпляром класса отличным от `Error` всегда возвращается 500.
Новые классы ошибок как раз создавались для того, чтобы был единый принцип работы. Вы здесь забыли про то, что в этих классах уже указан код ответа, нужно его использовать.
Также рекомендую для 500 отправлять просто какой-то текст, типа `'Произошла ошибка'`.

```suggestion
  const statusCode = error.statusCode || 500;
  return res.status(statusCode).send({ 
    message: statusCode === 500 ? 'Произошла ошибка' : error.message,
  });
```